url4df616fce9_s3e0n1g1_prototype3_tgz-pJ8-Trading_Engine_MainJ8/src/gui/ResultDisplay.java

```diff
@@ -0,0 +1,735 @@
+					LineGraph.addToDataset2(finishTime, tmpPrice);
+				}
+			};
+			set.close();
+		}catch (SQLException e) {
+			System.out.println("In Mainmenu/insertGraph : " + e);
+		}
+		
+		//finalise dataset for graph
+		returntimegraph.finishGraph(true);
+		returntimegraph.setVisible(true);
+		graphpanel.add(returntimegraph);
+
+		//Configurations for graph, not sure if can be done
+
+		JPanel config = new JPanel();
+		config.setLayout((new BoxLayout(config, BoxLayout.PAGE_AXIS)));
+		config.add(new JLabel("Configure Display"));
+		JButton updategraph = new JButton("Update");
+		updategraph.setSize(new Dimension(150,75));
+		config.add(updategraph);
+		config.setBorder(new EmptyBorder(0, 0, 0, 100)) ;
+		c.fill = GridBagConstraints.HORIZONTAL;
+		c.weightx = 0.5;
+		c.gridx = 0;
+		c.gridy = 0;
+		panel.add(graphpanel,c );
+		c.gridx = 2;
+		c.gridy = 1;
+		c.fill = GridBagConstraints.VERTICAL;
+		c.weightx = 2;
+
+		return panel;
+	}
+	
+	private StrategySelected myStrategyResult; 
+	//IMPLEMENT TRADING STRATEGY HERE
+
+	protected void runReversionStrategy() {
+		try{
+			ResultSet set = myDB.getResultSet("SELECT * FROM all_list;");
+			MyBidList myBidList = new MyBidList();
+			MyAskList myAskList = new MyAskList();
+			completedTrade = new LinkedList<ResultData>();
+			strategyResult = new LinkedList<String>();
+			askFirstList = new LinkedList<GraphData>();
+			bidFirstList = new LinkedList<GraphData>();
+			strategyAsk = new LinkedList<GraphData>();
+			strategyBid = new LinkedList<GraphData>();
+			LinkedList<ResultData> strategyTrade = new LinkedList<ResultData>();
+			int count = 0;
+			String tmp;
+			String tmpType;
+			double tmpPrice;
+			int tmpVol;
+			long tmpID = 0;
+			int updateLines = 0;
+			int deleteLines = 0;
+			int strategyCount = 0;
+			int tmpCount = 0;
+			Time tmpTime;
+			Time startTime = Time.valueOf("10:00:00");
+			Time endTime = Time.valueOf("16:00:00");
+			LinkedList<signalObject> signalList;
+			signalObject tmpSignal;
+			LinkedList<Long> buyOrderID = new LinkedList<Long>();
+			LinkedList<Long> sellOrderID = new LinkedList<Long>();
+			LinkedList<Long> OrderID;
+			lecMSReverse strategy = new lecMSReverse();
+			strategy.setThreShold(Threshold);
+			strategy.setAmountToTrade(stockAmount); 
+
+			long currentID = 0;
+			while (set.next()){
+				tmp = set.getString(5);
+				tmpType = set.getString(14);
+				tmpTime = set.getTime(3);
+				if(tmp.equalsIgnoreCase("ENTER")){
+					tmpPrice = set.getDouble(6);
+					tmpVol = set.getInt(7);
+					if(tmpType.equalsIgnoreCase("B")){
+						tmpID = set.getLong(12);
+						myBidList.add(tmpID,tmpPrice,tmpVol,tmpTime);
+						bidFirstList.add(new GraphData(myBidList.get(0).getPrice(),tmpTime));
+					}else if(tmpType.equalsIgnoreCase("A")){
+						tmpID = set.getLong(13);
+						myAskList.add(tmpID,tmpPrice,tmpVol,tmpTime);
+						askFirstList.add(new GraphData(myAskList.get(0).getPrice(),tmpTime));
+					}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 358:366, 513:521.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.